examples/sqlite/src/entities/posts/db.ts

```diff
@@ -0,0 +1,100 @@
+import { Database, SqlValue } from 'sql.js/dist/sql-asm.js';
+
+import { sqlite } from '../../db';
+import { User } from '../users/db';
+
+type Post = {
+    id: number;
+    title: string;
+    body: string;
+    user: User;
+};
+type PostCreate = Omit<Post, 'id' | 'user'> & { user_id: number };
+type PostUpdate = Pick<Post, 'id'> & Partial<PostCreate>;
+
+export function getPosts(db: Database, limit: number, offset: number): Post[] {
+    return sqlite<Post>`SELECT * FROM posts JOIN users ON posts.user_id = users.id ORDER BY posts.id LIMIT ${limit} OFFSET ${offset}`(
+        db,
+        convertPost
+    );
+}
+
+export function getPost(db: Database, id: number): Post | null {
+    const posts =
+        sqlite<Post>`SELECT * FROM posts JOIN users ON posts.user_id = users.id WHERE posts.id = ${id}`(
+            db,
+            convertPost
+        );
+
+    return posts.length === 0 ? null : posts[0];
+}
+
+export function countPosts(db: Database): number {
+    const results =
```

Comment:
Can't you just do this? (Same question as in countUsers)
```sql
SELECT COUNT(*)
FROM posts
```

Reply:
Does a full table scan and runs out of cycles if I remember correctly. It wasn't straight-forward to get a count in raw sql

Reply:
Done